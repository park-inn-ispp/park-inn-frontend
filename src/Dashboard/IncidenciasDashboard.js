import React, { useEffect, useState} from 'react';
import Loading from '../components/Loading';
import { useNavigate } from 'react-router-dom';
import call from '../Util/Caller'

export default function IncidenciasDashboard(){
    let navigate = useNavigate();
    const [incidencias, setIncidencias] = useState([]);
    const [isLoading, setIsLoading] = useState(true);
    var totalIncidencias = Object.keys(incidencias).length;
    var abiertas = Object.keys(incidencias.filter(x => x.estado==="abierta")).length;
    var cerradas = Object.keys(incidencias.filter(x => x.estado==="cerrada")).length;
    useEffect(() => {
        const Dashboard = async () => {
            const data = await call('/incidencias/all', 'GET');
            const incidencias = await data.json()
            setIncidencias(incidencias);
            setIsLoading(false);
        }
        Dashboard();
    });


    function cerrarIncidencia(id) {
        const data= {
            "estado":"cerrado"
        }
        call(`/incidencias/`+id, 'PUT', data)
          .then(response => {
            if (response.ok){
              navigate(`/dashboard-incidencias`)
            }
          })
      }

    if (isLoading) {
        return <Loading/>;
      }
      return (
        <div className='tablas'>
            <table >
                <tr>
                    <th>Id</th>
                    <th>Título</th>
                    <th>Descripción</th>
                    <th>Fecha</th>
                    <th>Email</th>
                    <th>IdReserva</th>
                    <th>Estado</th>
                    <th>Acciones</th>
                </tr>
                {incidencias.map((incidencia) => {
                    return <tr>
                        <td>{incidencia.id}</td>
                        <td>{incidencia.titulo}</td>
                        <td>{incidencia.descripcion}</td>
                        <td>{incidencia.fecha}</td>
                        <td>{incidencia.email}</td>
                        <td>{incidencia.idReserva}</td>
                        <td>{incidencia.estado}</td>
                        <td><button type='button' class='deleteButton' onClick={() => cerrarIncidencia(incidencia.id)}>Cerrar Incidencia</button></td>

                    </tr>
                })
            }
            </table>
            <div className='form-style-10'>
    <table>
    <tr>
        <tr>
        <td scope='row' abbr='numIncidencias'>Número total de incidencias realizadas</td>
        <td>{totalIncidencias}</td>
        </tr>
        <tr>
            <td scope='row' abbr='numIncidenciasAbiertas'>Número total de incidencias abiertas</td>
            <td>{abiertas}</td>
        </tr>
        <tr>
            <td scope='row' abbr='numIncidenciasCerradas'>Número total de incidencias cerradas</td>
            <td>{cerradas}</td>
        </tr>
    </tr>
    </table>
    </div>
        </div>
)
}