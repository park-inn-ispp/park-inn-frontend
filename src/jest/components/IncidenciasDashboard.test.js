import IncidenciasDashboard from "../../Dashboard/IncidenciasDashboard";
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({adapter: new Adapter()});

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ rates: { CAD: 1.42 } }),
  })
);

test('Incidencias dashboard empty', () => {

    const dashboard = <IncidenciasDashboard />;
     const tree = renderer
     .create(dashboard)
     .toJSON();

    console.log(tree)
    expect(tree).toMatchSnapshot();
});