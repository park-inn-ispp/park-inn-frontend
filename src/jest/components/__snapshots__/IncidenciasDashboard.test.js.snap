// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incidencias dashboard empty 1`] = `
<div
  className="loading-div"
>
  <div
    className="loading-gif"
  >
    <div
      data-testid="tail-spin-loading"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <svg
        aria-label="loading"
        data-testid="tail-spin-svg"
        height={80}
        viewBox="0 0 38 38"
        width="100vw"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <linearGradient
            id="a"
            x1="8.042%"
            x2="65.682%"
            y1="0%"
            y2="23.865%"
          >
            <stop
              offset="0%"
              stopColor="#011a30"
              stopOpacity="0"
            />
            <stop
              offset="63.146%"
              stopColor="#011a30"
              stopOpacity=".631"
            />
            <stop
              offset="100%"
              stopColor="#011a30"
            />
          </linearGradient>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(1 1)"
          >
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              id="Oval-2"
              stroke="#011a30"
              strokeWidth="2"
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
            <circle
              cx="36"
              cy="18"
              fill="#fff"
              r={1}
            >
              <animateTransform
                attributeName="transform"
                dur="0.9s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
